include/cuda/std/detail/libcxx/include/__expected/expected_base.h

```diff
@@ -0,0 +1,890 @@
+  using __base::__base;
+};
+
+template <class _Tp, class _Err>
+struct __expected_move_assign<_Tp, _Err, false> : __expected_copy_assign<_Tp, _Err>
+{
+  using __base = __expected_copy_assign<_Tp, _Err>;
+  using __base::__base;
+
+  __expected_move_assign() = default;
+  __expected_move_assign(const __expected_move_assign&) = default;
+  __expected_move_assign(__expected_move_assign&&) = default;
+  __expected_move_assign& operator=(const __expected_move_assign&) = default;
+
+  _LIBCUDACXX_INLINE_VISIBILITY _LIBCUDACXX_CONSTEXPR_AFTER_CXX17
+  __expected_move_assign& operator=(__expected_move_assign&& __other)
+    noexcept(_LIBCUDACXX_TRAIT(is_nothrow_move_assignable, _Tp) &&
+             _LIBCUDACXX_TRAIT(is_nothrow_move_constructible, _Tp) &&
+             _LIBCUDACXX_TRAIT(is_nothrow_move_assignable, _Err) &&
+             _LIBCUDACXX_TRAIT(is_nothrow_move_constructible, _Err)) // strengthened
+  {
+    if (this->__has_val_ && __other.__has_val_) {
+      this->__union_.__val_ = _CUDA_VSTD::move(__other.__union_.__val_);
+    } else if (this->__has_val_ && !__other.__has_val_) {
+      this->__reinit_expected(this->__union_.__unex_, this->__union_.__val_, _CUDA_VSTD::move(__other.__union_.__unex_));
+      this->__has_val_ = false;
+    } else if (!this->__has_val_ && __other.__has_val_) {
+      this->__reinit_expected(this->__union_.__val_, this->__union_.__unex_, _CUDA_VSTD::move(__other.__union_.__val_));
+      this->__has_val_ = true;
+    } else { // !this->__has_val_ && !__other.__has_val_
+      this->__union_.__unex_ = _CUDA_VSTD::move(__other.__union_.__unex_);
+    }
+    return *this;
+  }
+};
+
+template <class _Tp, class _Err>
+using __expected_sfinae_ctor_base_t = __sfinae_ctor_base<
+  _LIBCUDACXX_TRAIT(is_copy_constructible, _Tp) && _LIBCUDACXX_TRAIT(is_copy_constructible, _Err),
+  _LIBCUDACXX_TRAIT(is_move_constructible, _Tp) && _LIBCUDACXX_TRAIT(is_move_constructible, _Err)
+>;
+
+template <class _Tp, class _Err>
+using __expected_sfinae_assign_base_t = __sfinae_assign_base<
+  _LIBCUDACXX_TRAIT(is_copy_constructible, _Tp) && _LIBCUDACXX_TRAIT(is_copy_constructible, _Err) &&
+  _LIBCUDACXX_TRAIT(is_copy_assignable, _Tp)    && _LIBCUDACXX_TRAIT(is_copy_assignable, _Err) &&
+  (_LIBCUDACXX_TRAIT(is_nothrow_move_constructible, _Tp) || _LIBCUDACXX_TRAIT(is_nothrow_move_constructible, _Err)),
+  _LIBCUDACXX_TRAIT(is_move_constructible, _Tp) && _LIBCUDACXX_TRAIT(is_move_constructible, _Err) &&
+  _LIBCUDACXX_TRAIT(is_move_assignable, _Tp)    && _LIBCUDACXX_TRAIT(is_move_assignable, _Err) &&
+ (_LIBCUDACXX_TRAIT(is_nothrow_move_constructible, _Tp) || _LIBCUDACXX_TRAIT(is_nothrow_move_constructible, _Err))
+>;
+
+// expected<void, E> base classtemplate <class _Tp, class _Err>
+// MSVC complains about [[no_unique_address]] prior to C++20 as a vendor extension
+#if defined(_LIBCUDACXX_COMPILER_MSVC)
+#pragma warning(push)
+#pragma warning(disable : 4848)
+#endif // _LIBCUDACXX_COMPILER_MSVC
```

Comment:
Do we still need this with the config macro for this attribute?

Reply:
Yes we need it because the macro falls back to `[[msvc::no_unique_address]]` which leads to that warning I am not sure if there is anything inside the macro we can do to disable the warning